### PR TITLE
US-498622 - Infinity OAuth env vars for SRS connector

### PR DESCRIPTION
**Pull request by wieslaw-bartz**
As SRS supports IDP based authentication/authorization, in CMC, client can now enable authentication between Infinity and SRS by setting 'pegasearch.srsAuth.enabled'  flag to true and configure other settings in 'pegasearch.srsAuth' with OAuth client parameters from IDP

### CODE REVIEW

**Review by pega-talba — APPROVED** [review]

**Review by APegaDavis — APPROVED** [review]

File: charts/pega/README.md [hunk 1]
```diff
@@ -801,6 +801,32 @@ pegasearch:
   externalURL: "http://srs-service.namespace.svc.cluster.local"
 ```
 
+To use OAuth authorization service for the connection between Pega Infinity and Search and Reporting Service, follow the instructions provided at ['backingservices'](../backingservices) to configure connection to the authorization service in Search and Reporting Service. You must configure authorization parameters for your Pega Infinity deployment using the parameter in values.yaml as shown in the following table and example:
```
Comment (taz-pega-work) [inline charts/pega/README.md:804]: can we tweak the wording a little so the context is more clear. Sorry that the original readme wording wasn't that clear...but this should help!

To configure authorization for the connection between Pega Infinity and the Search and Reporting Service (SRS)  use the OAuth authorization service. For more information, see ['backingservices'](../backingservices). To configure the connection to the authorization service in SRS you must configure the following authorization parameters in the Pega values.yaml as shown in the following table and example:

File: charts/pega/README.md [hunk 2]
```diff
@@ -801,6 +801,32 @@ pegasearch:
...
+
+| Parameter             | Description                                                                                                                                                                                                 | Default value        |
+|-----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------|
+| `enabled`             | Set the `pegasearch.srsAuth.enabled` as true to use OAuth between Infinity and Search and Reporting service                                                                                                 | false                |
```
Comment (taz-pega-work) [inline charts/pega/README.md:808]: Set the `pegasearch.srsAuth.enabled` to 'true' to use OAuth between Infinity and SRS.

File: charts/pega/README.md [hunk 3]
```diff
@@ -801,6 +801,32 @@ pegasearch:
...
+| `url`                 | Set the `pegasearch.srsAuth.url` value to URL of the OAuth endpoint to get the token for Search and Reporting service                                                                                       | `""`                 |
```
Comment (taz-pega-work) [inline charts/pega/README.md:809]: Set the `pegasearch.srsAuth.url` value to the URL of the OAuth service endpoint to get the token for SRS.

File: charts/pega/README.md (hunk 2 above)
Comment (wieslaw-bartz) [reply charts/pega/README.md]: Done

File: charts/pega/README.md (hunk 3 above)
Comment (wieslaw-bartz) [reply charts/pega/README.md]: Done

File: charts/pega/README.md [hunk 4]
```diff
@@ -801,6 +801,32 @@ pegasearch:
...
+| `clientId`            | Set the `pegasearch.srsAuth.clientId` value to the client id used in OAuth service                                                                                                                          | `""`                 |
```
[the remark this answers is not in the document]
Comment (wieslaw-bartz) [reply charts/pega/README.md]: Done

File: charts/pega/README.md [hunk 5]
```diff
@@ -801,6 +801,32 @@ pegasearch:
...
+| `scopes`              | Set the `pegasearch.srsAuth.scopes` value to "pega.search:full". It is the scope in OAuth service required to grant access to Search And Reporting Service                                                  | `"pega.search:full"` |
+| `privateKey`          | Set the `pegasearch.srsAuth,privateKey` value to OAuth private PKCS8 key (additionally encoded with base64) used to get an authorization token for Pega Infinity connection to Search and Reporting Service | `""`                 |
```
[the remark this answers is not in the document]
Comment (wieslaw-bartz) [reply charts/pega/README.md]: Done

File: charts/pega/README.md [hunk 6]
```diff
@@ -801,6 +801,32 @@ pegasearch:
...
+| `privateKeyAlgorithm` | Set the `pegasearch.srsAuth.privateKeyAlgorithm` value to Algorithm used to generate a private key used by OAuth client. Allowed values: RS256, RS384, RS512, ES256, ES384, ES512. Default value: RS256     | `"RS256"`            |
```
[the remark this answers is not in the document]
Comment (wieslaw-bartz) [reply charts/pega/README.md]: @taz-pega-work 
I have updated it too.
Please review it.

**Review by taz-pega-work — APPROVED** [review]
looks great! nice work. thanks for collaborating!

**Review by Heshma — APPROVED** [review]

**Pull request merged**